components/script/dom/document.rs

```diff
@@ -4312,10 +4312,13 @@ impl Document {
                 return csp::CheckResult::Allowed;
             },
             Some(csp_list) => {
+                // <https://www.w3.org/TR/CSP/#is-element-nonceable>
                 let element = csp::Element {
+                    // Step 1: If element does not have an attribute named "nonce", return "Not Nonceable".
                     nonce: el
                         .get_attribute(&ns!(), &local_name!("nonce"))
-                        .map(|attr| Cow::Owned(attr.value().to_string())),
+                        .and_then(|attr| el.is_element_nonceable(attr))
+                        .map(|ref_str| Cow::Owned(ref_str)),
```

Comment:
Is there a particular reason why step 1 can't be in `Element::is_element_nonceable`?
Seems to me like you could just do 

```rust
// Step 1: If element does not have an attribute named "nonce", return "Not Nonceable".
let nonce_attribute = self.get_attribute(&ns!(), &local_name!("nonce"))?;
```

at the beginning of the function